DependencyInjection/Compiler/AddProviderCompilerPass.php

```diff
@@ -40,31 +42,21 @@ public function process(ContainerBuilder $container)
     }
 
     /**
+     * NEXT_MAJOR: Remove this method.
+     *
      * @param ContainerBuilder $container
      *
      * @return array
      */
     public function fixSettings(ContainerBuilder $container)
     {
-        $pool = $container->getDefinition('sonata.media.pool');
-
-        // not very clean but don't know how to do that for now
-        $settings = false;
-        $methods = $pool->getMethodCalls();
-        foreach ($methods as $pos => $calls) {
-            if ($calls[0] == '__hack__') {
-                $settings = $calls[1];
-                break;
-            }
-        }
+        @trigger_error(
+            'The '.__METHOD__.' method is deprecated since 3.x, to be removed in 4.0. '.
+            'Use AddProviderCompilerPass::getExtensionConfig() instead.',
```

Comment:
Should I declare it `protected` ? 

Maybe we can make it private on `master`and treat it like "Closing API" or make this class final